File: 1524-number-of-sub-arrays-with-odd-sum/1524-number-of-sub-arrays-with-odd-sum.c


int numOfSubarrays(int* arr, int arrSize){
    int evenCount = 0, oddCount = 0;
    int mod = 1e9 + 7;
    
    if(arr[0] % 2 == 0)
        ++evenCount;
    else
        ++oddCount;
    int ans = oddCount;
    
    for(int i = 1; i < arrSize; ++i) {
        if(arr[i] % 2 == 0)
            ++evenCount;
        else {
            int tmp = oddCount;
            oddCount = evenCount + 1;
            evenCount = tmp;
        }
        ans = (ans + oddCount) % mod;
    }
    return ans;
}

File: 1524-number-of-sub-arrays-with-odd-sum/1524-number-of-sub-arrays-with-odd-sum.py
class Solution:
    def numOfSubarrays(self, arr: List[int]) -> int:
        evenCount = 0
        oddCount = 0
        
        if arr[0] % 2 == 0:
            evenCount = 1
        else:
            oddCount = 1
        
        ans = oddCount
        mod = int(1e9 + 7)
        
        for i in range(1, len(arr)):
            if arr[i] % 2 == 0:
                evenCount += 1
            else:
                tmp = oddCount
                oddCount = evenCount + 1
                evenCount = tmp
            ans = (ans + oddCount) % mod
        
        return ans
        